lib/ansible/modules/crypto/luks_device.py

```diff
@@ -0,0 +1,512 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: luks_device
+
+short_description: Manage encrypted (LUKS) devices
+
+version_added: "2.8"
+
+description:
+    - "Module manages L(LUKS,https://en.wikipedia.org/wiki/Linux_Unified_Key_Setup)
+      on given device. Supports creating, destroying, opening and closing of
+      LUKS container and adding or removing new keys."
+
+options:
+    device:
+        description:
+            - "Device to work with (e.g. C(/dev/sda1)). Needed in most cases.
+              Can be omitted only when I(state=closed) together with I(name)
+              is provided."
+        type: str
+    state:
+        description:
+            - "Desired state of the LUKS container. Based on its value creates,
+              destroys, opens or closes the LUKS container on a given device."
+            - "I(present) will create LUKS container unless already present.
+              Requires I(device) and I(keyfile) options to be provided."
+            - "I(absent) will remove existing LUKS container if it exists.
+              Requires I(device) or I(name) to be specified."
+            - "I(opened) will unlock the LUKS container. If it does not exist
+              it will be created first.
+              Requires I(device) and I(keyfile) to be specified. Use
+              the I(name) option to set the name of the opened container.
+              Otherwise the name will be generated automatically and returned
+              as a part of the result."
+            - "I(closed) will lock the LUKS container. However if the container
+              does not exist it will be created.
+              Requires I(device) and I(keyfile) options to be provided. If
```

Comment:
Closing does not need `keyfile`. (Tested it with the module, works fine without `keyfile` option.)

Reply:
That is technically true. However `closed` can now also create a LUKS container if it does not exist. And container creation requires `device` and `keyfile`. In case the user is absolutely sure that container is already present only `device` or even just `name` will do. Next sentence in the doc explains that.

Reply:
Ah, I misread that. Totally makes sense :)